tests/mu_convert_enum_test.c

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <string.h>
+#include "test_helper.h"
+
+#include "azure_macro_utils/macro_utils.h"
+
+#include "mu_convert_enum_test.h"
+
+// This lists the values in the first enum
+#define TEST_CONVERT_FROM_ENUM_VALUES \
+    test_from_a, \
+    test_from_b, \
+    test_from_c
+
+#define TEST_CONVERT_TO_ENUM_VALUES \
+    test_to_a, \
+    test_to_b
+
+MU_DEFINE_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_FROM_ENUM_VALUES);
+MU_DEFINE_ENUM(TEST_CONVERT_TO_ENUM, TEST_CONVERT_TO_ENUM_VALUES);
+
+/* The following code should not compile :
```

Comment:
>should not compile : [](start = 22, length = 20)

"the reason is left as an exercise to the reader".

So I took the branch... aaaaaand.... surprise! It does compile (after commenting obviously lines 29-32)

Reply:
Probably should be `MU_DEFINE_CONVERT_ENUM_WITH_VALIDATION` ?

---
In reply to: [394386477](https://github.com/Azure/azure-macro-utils-c/pull/36#discussion_r394386477) [](ancestors = 394386477)

Reply:
Added "why" and now that the MU_DEFINE_CONVERT_ENUM has validation by default indeed it does not compile.

---
In reply to: [394484403](https://github.com/Azure/azure-macro-utils-c/pull/36#discussion_r394484403) [](ancestors = 394484403,394386477)